### PR TITLE
[HDInsight]-Add disk encryption related cmdlet for hdi cluster

### PR DESCRIPTION
**Pull request by sunsw1994**
## Description

1. Register HDInsight as a managed identity with Azure Key Vault and add the encryption key when create HDInsight cluster.
2. Change the encryption keys used by the HDInsight cluster after it has been created

3. [Design Review
](https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/578)

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

File: src/HDInsight/HDInsight/ManagementCommands/SetAzureHDInsightClusterDiskEncryptionKeyCommand.cs [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.HDInsight.Models;
+using System.Linq;
+using System.Management.Automation;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+
+namespace Microsoft.Azure.Commands.HDInsight
+{
+    [Cmdlet("Set", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "HDInsightClusterDiskEncryptionKey"),OutputType(typeof(Cluster))]
```
Comment (idear1203) [inline src/HDInsight/HDInsight/ManagementCommands/SetAzureHDInsightClusterDiskEncryptionKeyCommand.cs:26]: >Cmdlet [](start = 5, length = 6)

Nice to add a default parameter set.

File: src/HDInsight/HDInsight/help/Set-AzHDInsightClusterDiskEncryptionKey.md [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+---
+
+# Set-AzHDInsightClusterDiskEncryptionKey
+
+## SYNOPSIS
+Rotate the disk encryption key of the specified HDInsight cluster.
+
+## SYNTAX
+
+### SetByNameParameterSet
```
Comment (aim-for-better) [inline src/HDInsight/HDInsight/help/Set-AzHDInsightClusterDiskEncryptionKey.md:16]: We need to add a default parameter set in command and regenerate the help doc.

File: src/HDInsight/HDInsight/ManagementCommands/SetAzureHDInsightClusterDiskEncryptionKeyCommand.cs (hunk 1 above)
Comment (aim-for-better) [reply src/HDInsight/HDInsight/ManagementCommands/SetAzureHDInsightClusterDiskEncryptionKeyCommand.cs:26]: Please remember to update the help doc after adding the default parameter set

---
In reply to: [417792081](https://github.com/Azure/azure-powershell/pull/11761#discussion_r417792081) [](ancestors = 417792081)

File: src/HDInsight/HDInsight/ManagementCommands/NewAzureHDInsightClusterCommand.cs [hunk 3]
```diff
@@ -358,6 +368,23 @@ public DateTime RdpAccessExpiry
...
+        [Parameter(HelpMessage = "Gets or sets the encryption key name.")]
+        public string EncryptionKeyName { get; set; }
+
+        [Parameter(HelpMessage = "Gets or sets the encryption key version.")]
+        public string EncryptionKeyVersion { get; set; }
+
+        [Parameter(HelpMessage = "Gets or sets the encryption vault uri.")]
+        public string EncryptionVaultUri { get; set; }
+
+
```
Comment (aim-for-better) [inline src/HDInsight/HDInsight/ManagementCommands/NewAzureHDInsightClusterCommand.cs:387]: Nice to keep one empty line just like before

**Comment by aim-for-better** [comment]
>       "RequestUri": "/subscriptions/964c10bb-8a6c-43bc-83d3-6b318c6c7305/resourcegroups/group-ps-test2596?api-version=2016-09-01",

Nice to replace with fake data

---
Refers to: src/HDInsight/HDInsight.Test/SessionRecords/Commands.HDInsight.Test.ScenarioTests.HDInsightClusterTests/TestCmkClusterRelatedCommands.json:4 in b49e337. [](commit_id = b49e337ca10a59c13c23c2df59226e7058308a51, deletion_comment = False)

File: src/HDInsight/HDInsight.Test/ScenarioTests/TestController.cs [hunk 4]
```diff
@@ -74,7 +80,7 @@ public void RunPowerShellTest(XunitTracingInterceptor logger, params string[] sc
                 mockName);
         }
 
-        public void RunPsTestWorkFlow(Func<string[]> scriptBuilder, Action cleanup, string callingClassType, string mockName)
+        public void RunPsTestWorkFlow(Func<string[]> scriptBuilder, System.Action cleanup, string callingClassType, string mockName)
```
Comment (aim-for-better) [inline src/HDInsight/HDInsight.Test/ScenarioTests/TestController.cs:83]: >System. [](start = 68, length = 7)

Is it necessary?

Comment (sunsw1994) [reply src/HDInsight/HDInsight.Test/ScenarioTests/TestController.cs:83]: Yes. It conflicts with other dll.

Comment (aim-for-better) [reply src/HDInsight/HDInsight.Test/ScenarioTests/TestController.cs:83]: Ok, Got it

---
In reply to: [417800100](https://github.com/Azure/azure-powershell/pull/11761#discussion_r417800100) [](ancestors = 417800100)

**Pull request merged**